rfcs/testdriver-features.md

```diff
@@ -0,0 +1,64 @@
+# RFC 214: Add testdriver features
+
+## Summary
+
+This RFC is an alternative to [RFC 212](https://github.com/web-platform-tests/rfcs/pull/212) providing a more generic and extendable solution. It proposes adding a query parameter `?feature` to `testdriver.js` and adding a `testdriver_features` property to the [TestharnessTest](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-8f280b64b0700ab9b4b343adabc7ff4d5ea4b1f45cb6c4bd7f50a19b21ebdb8cR169). At the moment the only supported feature will be `bidi`, which will be used to enable the [testdriver BiDi API](https://github.com/web-platform-tests/rfcs/blob/master/rfcs/testdriver_bidi.md) for specific WPT tests. The feature `bidi`, applicable to HTML and JS files, would let test runners choose between [**WebDriverProtocol**](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-c2f15328bc1ddfa8fb93c09ae41651e2bcfdad4f257932263a3e92c3f8deffceR277) and [**WebDriverBidiProtocol**](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-c2f15328bc1ddfa8fb93c09ae41651e2bcfdad4f257932263a3e92c3f8deffceR279). This approach does not require all the browsers to implement WebDriver BiDi.
+
+The query parameter `feature` currently supports only the `bidi` feature. Future implementations may allow for multiple `feature` parameters, such as `testdriver.js?feature=bidi&feature=SOME_OTHER_FEATURE`.
```

Comment:
```suggestion
The query parameter `feature` takes a single value, which is a string representing a feature name. Feature names are compared by case-sensitive string matching. Multiple features are represented as multiple `feature` parameters for example `testdriver.js?feature=bidi&feature=SOME_OTHER_FEATURE`. This RFC only defines the `bidi` feature; future RFCs may add additional features.
```

Reply:
I would also suggest that we explicitly state that any features added by vendors for their own tests must contain a `:` in the name; then we can write a lint that only permitted features are being used and so catch `feature=BiDi` or similar.

Reply:
> I would also suggest that we explicitly state that any features added by vendors for their own tests must contain a `:` in the name; then we can write a lint that only permitted features are being used and so catch `feature=BiDi` or similar.

I believe it makes sense not only to allow for extension features, but for extension query parameters, which is implemented in the prototype lint rules for [query parameters](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-b3de46e07408c6a68229472f80ec0f23d8a7ff634bc01b25f51ff054478dcc60R554) and [features](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-b3de46e07408c6a68229472f80ec0f23d8a7ff634bc01b25f51ff054478dcc60R562).

Reply:
Added this description to the RFC.

Reply:
> I believe it makes sense not only to allow for extension features, but for extension query parameters, which is implemented in the prototype lint rules for [query parameters](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-b3de46e07408c6a68229472f80ec0f23d8a7ff634bc01b25f51ff054478dcc60R554) and [features](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-b3de46e07408c6a68229472f80ec0f23d8a7ff634bc01b25f51ff054478dcc60R562).

After some further consideration, I don't think we should allow for any vendor-specific query parameters at least for now. If required, this can be changed in another RFC.